sway-core/src/semantic_analysis/namespace/module.rs

```diff
@@ -285,6 +298,75 @@ impl Module {
         let parent_scope_id = self.current_lexical_scope().parent;
         self.current_lexical_scope_id = parent_scope_id.unwrap_or(0);
     }
+
+    pub fn get_items_for_type(
+        &self,
+        engines: &Engines,
+        type_id: TypeId,
+    ) -> Vec<ResolvedTraitImplItem> {
+        let mut lexical_scope_opt = Some(self.current_lexical_scope());
+        let mut vec = vec![];
+        while let Some(lexical_scope) = lexical_scope_opt {
+            vec.extend(
+                lexical_scope
+                    .items
+                    .implemented_traits
+                    .get_items_for_type(engines, type_id),
+            );
+            if let Some(parent_scope_id) = lexical_scope.parent {
+                lexical_scope_opt = self.get_lexical_scope(parent_scope_id);
+            } else {
+                lexical_scope_opt = None;
+            }
+        }
+        vec
+    }
+
+    pub fn resolve_symbol(
+        &self,
+        handler: &Handler,
+        engines: &Engines,
+        symbol: &Ident,
+    ) -> Result<ResolvedDeclaration, ErrorEmitted> {
+        let mut lexical_scope_opt = Some(self.current_lexical_scope());
+        while let Some(lexical_scope) = lexical_scope_opt {
+            let result = lexical_scope
+                .items
+                .resolve_symbol(handler, engines, symbol)?;
+            if let Some(result) = result {
+                return Ok(result);
+            }
+            if let Some(parent_scope_id) = lexical_scope.parent {
+                lexical_scope_opt = self.get_lexical_scope(parent_scope_id);
+            } else {
+                lexical_scope_opt = None;
+            }
+        }
+
+        // Symbol not found
+        Err(handler.emit_err(CompileError::SymbolNotFound {
+            name: symbol.clone(),
+            span: symbol.span(),
+        }))
+    }
+
+    pub fn get_methods_for_type(
+        &self,
+        engines: &Engines,
+        type_id: TypeId,
+    ) -> Vec<ResolvedFunctionDecl> {
+        self.get_items_for_type(engines, type_id)
+            .into_iter()
+            .filter_map(|item| match item {
+                ResolvedTraitImplItem::Parsed(_) => todo!(),
```

Comment:
nit: I see that this was just moved from another file, but I think we should replace the `todo!` with something else, like `None`